client/src/Components/UpdateTask.js

```diff
@@ -0,0 +1,104 @@
+
+import React, { useEffect, useState } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+ 
+
+export default function UpdateTask() {
+  const navigate = useNavigate();
+  const [errors,setErrors]=useState([])
+  const { id } = useParams();
+  const [title, setTitle] = useState("");
+  const [description, setDescription] = useState("");
+
+  useEffect(()=>{
+    fetch("http://localhost:5500/api/task/getTask/"+id,{
+        headers:{
+            "Content-Type":"application/json",
+            "auth-token":"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjp7ImlkIjoiNjYyZjYxMWM2MjRjYmNkNGUxMGNkZTdkIiwicm9sZSI6ImFkbWluIn0sImlhdCI6MTcxNDM4NjE5NX0.3WJQOYttPA_Hk202uI9WeLi8ejqzeHsqevHV_b2kCik"
+        }
+    }).then(response=>response.json())
+    .then(data=>{
+         console.log(data.title)
+        setTitle(data.title)
+        setDescription(data.description)
+    }).catch(error=>console.log(error))
+
+  },[])
+
+
+  const Submit = (e) => {
+    console.log(title, description);
+    e.preventDefault();
+   
+    fetch("http://localhost:5500/api/task/updateTask/" + id, {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+        "auth-token":
+          "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjp7ImlkIjoiNjYyZjYxMWM2MjRjYmNkNGUxMGNkZTdkIiwicm9sZSI6ImFkbWluIn0sImlhdCI6MTcxNDM4NjE5NX0.3WJQOYttPA_Hk202uI9WeLi8ejqzeHsqevHV_b2kCik",
+      },
+      body: JSON.stringify({
+        title: title,
+        description: description,
+      }),
+    })
+    .then((res) => {
+      if (!res.ok) {
+        return res.json().then((data) => {
+          throw { status: res.status, errors: data.errors }; // Throw error with status and errors
+        });
+      }
+      return res.json();
+    })
+    .then((data) => {
+      console.log("Data:", data);
+      navigate(`/getUser/${data.user}`);
+    })
+    .catch((error) => {
+       
+      if (error.status === 400) {
+        console.log("Validation Errors:", setErrors(error.errors));
+        // Handle validation errors here
+      }
+      else{
+        console.log("Error",error.message)
+      }
+    });
+  };
+
+  return (
+    <div className="container mt-5 py-5">
+      <div className="row justify-content-center py-2 border">
+        <div className="col-6">
+          <form onSubmit={Submit}>
+            <div className="mb-3">
+              <label htmlFor="title" className="form-label">Title</label>
+              <input type="text" className="form-control" id="title" value={title}  onChange={(e) => setTitle(e.target.value)}/>
```

Comment:
Change handler to simple functions, don't create anonymust functions, everywhere.
Create a single handler for all same time of inputs with switch case, everywhere.